lib/oli_web/controllers/cognito_controller.ex

```diff
@@ -0,0 +1,151 @@
+defmodule OliWeb.CognitoController do
+  use OliWeb, :controller
+  import Oli.HTTP
+  import Oli.Utils
+  import Ecto
+  import Ecto.Changeset
+  require Logger
+
+  alias Oli.Repo
+  alias Oli.Accounts
+  alias Oli.Groups.CommunityAccount
+  alias Oli.Institutions.SsoJwks
+
+  def launch(conn, params) do
+    try do
+      with {:ok, valid_params} <- validate_query_params(params) do
+        jwk = get_jwk(Map.get(params, "id_token"))
+        jwk_from_pem = JOSE.JWK.from_pem(jwk.pem)
+        case JOSE.JWT.verify_strict(jwk_from_pem, ["RS256"], valid_params.id_token) do
+          {true, verified_jwt, _} ->
+            case Accounts.insert_or_update_lms_user(%{
```

Comment:
We're handling the successful case but we're not covering the case when this function returns `{:error, changeset}`